src/code/audio/audio_seqplayer.c

```diff
@@ -1,73 +1,2310 @@
+                    sound = AudioPlayback_InstrumentGetSound(instrument, vel);
+                    sameSound = (layer->sound == sound);
+                    layer->sound = sound;
+                    tuning = sound->tuning;
+                } else {
+                    layer->sound = NULL;
+                    tuning = 1.0f;
+                    if (instOrWave >= 0xC0) {
+                        layer->sound = &gAudioContext.synthesisReverbs[instOrWave - 0xC0].sound;
+                    }
+                }
+
+                temp_f2 = gPitchFrequencies[semitone2] * tuning;
+                temp_f14 = gPitchFrequencies[layer->portamentoTargetNote] * tuning;
+
+                switch (PORTAMENTO_MODE(*portamento)) {
+                    case PORTAMENTO_MODE_1:
+                    case PORTAMENTO_MODE_3:
+                    case PORTAMENTO_MODE_5:
+                        freqScale2 = temp_f2;
+                        freqScale = temp_f14;
+                        break;
+
+                    case PORTAMENTO_MODE_2:
+                    case PORTAMENTO_MODE_4:
+                        freqScale = temp_f2;
+                        freqScale2 = temp_f14;
+                        break;
+
+                    default:
+                        freqScale = temp_f2;
+                        freqScale2 = temp_f2;
+                        break;
+                }
+
+                portamento->extent = (freqScale2 / freqScale) - 1.0f;
+
+                if (PORTAMENTO_IS_SPECIAL(*portamento)) {
+                    speed = seqPlayer->tempo * 0x8000 / gAudioContext.tempoInternalToExternal;
+                    if (layer->delay != 0) {
+                        speed = speed * 0x100 / (layer->delay * layer->portamentoTime);
+                    }
+                } else {
+                    speed = 0x20000 / (layer->portamentoTime * gAudioContext.audioBufferParameters.updatesPerFrame);
+                }
+
+                if (speed >= 0x7FFF) {
+                    speed = 0x7FFF;
+                } else if (speed < 1) {
+                    speed = 1;
+                }
+
+                portamento->speed = speed;
+                portamento->cur = 0;
+                layer->freqScale = freqScale;
+                if (PORTAMENTO_MODE(*portamento) == PORTAMENTO_MODE_5) {
+                    layer->portamentoTargetNote = semitone;
+                }
+                break;
+            }
+
+            if (instrument != NULL) {
+                sound = AudioPlayback_InstrumentGetSound(instrument, semitone);
+                sameSound = (sound == layer->sound);
+                layer->sound = sound;
+                layer->freqScale = gPitchFrequencies[semitone2] * sound->tuning;
+            } else {
+                layer->sound = NULL;
+                layer->freqScale = gPitchFrequencies[semitone2];
+                if (instOrWave >= 0xC0) {
+                    layer->sound = &gAudioContext.synthesisReverbs[instOrWave - 0xC0].sound;
```

Comment:
I certainly hope so. Reverb is a deep web of difficult documentation, even in OoT. And the biggest difference between MM and OoT in the audio library is that MM expanded on the reverb code quite a bit. So there's a lot of difficult work to do in OoT, then it can be expanded to MM. I've begun my testing of reverb recently, but it'll be a while before this is fully understood I think